### PR TITLE
Add shaderdb tests for DemoteToHelper extension

### PR DESCRIPTION
**Pull request by vettoreldaniele**

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
Can one of the admins verify this patch?

File: llpc/test/shaderdb/extensions/ExtDemoteToHelper_TestDemote.frag [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+void main()
+{
+    if  (discardPixel != 0) {
+    	demote;      
+    }
+    fragColor = texture(image, texCoordIn);
+}
+
+// BEGIN_SHADERTEST
```
Comment (kuhar) [inline llpc/test/shaderdb/extensions/ExtDemoteToHelper_TestDemote.frag:20]: Can you put test comments at the top?

File: llpc/test/shaderdb/extensions/ExtDemoteToHelper_TestIsHelperInvocation.frag [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+void main()
+{
+    if  (discardPixel != 0) {
+    	demote;      
+    }
+    bool isHelper = helperInvocationEXT();
+    fragColor = isHelper ? vec4(1.0f) : vec4(0.0f);
+}
+
+// BEGIN_SHADERTEST
```
Comment (kuhar) [inline llpc/test/shaderdb/extensions/ExtDemoteToHelper_TestIsHelperInvocation.frag:18]: Also here

**Comment by vettoreldaniele** [comment]
V1: Move checks at the top of the file.

**Review by kuhar — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 54d4ecde2bfa4e0b860946e0eff9147e8f066d29
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c6950e03a4c97beadc88a8bf5bea3dc5aa2649ee</li>
</ul></details>

<details><summary>CTS tests (Failed: 1/214490)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 29753/45458 (65.5%)</li>
<li><details><summary>Failed: 1/45458 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.subgroups.clustered.compute.subgroupclusteredand_u8vec2
Stack trace: Script:
Crash

```
</p></details></li>

<li>Not Supported: 15703/45458 (34.5%)</li><li>Warnings: 1/45458 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Review by kuhar — DISMISSED** [review]

**Review by kuhar — APPROVED** [review]

**Review by amdrexu — APPROVED** [review]

**Comment by amdvlk-admin** [comment]
### Test summary for commit 2e4f817b4bbac95238c26988d0db9b47b5b4b898
<details><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master 39f8940199e60c44d4211cf8165dfd12876316fa</li>
<li>METROHASH: amd-master 3c566dd9cda44ca7fd97659e0b53ac953f9037d2</li>
<li>PAL: dev bcec463efe5260776d486a5e3da0c549bc0a75d2</li>
<li>SPVGEN: dev 580ade71d630a304805e9468ef22ad263c37d137</li>
<li>XGL: dev 04534321e450d220d404a240a047c7e1cd82f3fd</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c6950e03a4c97beadc88a8bf5bea3dc5aa2649ee</li>
</ul></details>

<details><summary>CTS tests (Failed: 0/225376)</summary><ul><li>Built with version 1.2.5.2</li>
<details open><summary>Rhel 8.2, Gfx10</summary><ul>

<li>Passed: 32541/56344 (57.8%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23803/56344 (42.2%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Gfx9</summary><ul>

<li>Passed: 32381/56344 (57.5%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23962/56344 (42.5%)</li><li>Warnings: 1/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx8</summary><ul>

<li>Passed: 32717/56344 (58.1%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 23627/56344 (41.9%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx103</summary><ul>

<li>Passed: 35698/56344 (63.4%)</li>
<li>Failed: 0/56344 (0.0%)</li>
<li>Not Supported: 20646/56344 (36.6%)</li><li>Warnings: 0/56344 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**